tests/tst_kdalgorithms.cpp

```diff
@@ -1161,6 +1165,133 @@ void TestAlgorithms::accumulate()
     }
 }
 
+void TestAlgorithms::accumulateAndMemberFunctions()
+{
+    struct ResultBuilder
+    {
+        ResultBuilder &append(const std::string &other)
+        {
+            result += "/" + other;
+            return *this;
+        }
+        std::string result;
+    };
+
+    {
+        std::vector<std::string> list{"abc", "def", "hij"};
+        ResultBuilder result;
+        (void)kdalgorithms::accumulate(list, &ResultBuilder::append, std::ref(result));
```

Comment:
Not a fan of the API here:

result = accumulate(list, &...append, ResultBuilder{})

would be cleaner IMO